compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -1767,6 +1767,19 @@ static void buildRecordQueryVarField(FnSymbol*  fn,
 *                                                                             *
 ************************************** | *************************************/
 
+static bool inheritsFromError(Type* t) {
+  if (t == dtError)
+    return true;
+
+  bool ret = false;
+
+  forv_Vec(Type, parent, t->dispatchParents) {
+    ret = ret || inheritsFromError(parent);
```

Comment:
Is it possible for there to be parents beyond `dtError` in `dispatchParents`? If so, maybe it'd be better to check if each parent is `dtError` and return true if it is, rather than going through the whole list.

Reply:
Or is it possible for a parent's `dispatchParents` to contain classes that are not in the child's `dispatchParents`? In that case this loop makes sense.

Reply:
yes, if you had FileNotFoundError inherit from SystemError inherit from Error, for example.